### DISCUSSION TITLE
Can I wait until some file exists on my project directory?

### DISCUSSION THREAD

**Discussion by ramesh-khatri** [getgauge/taiko#1809] [General]
I am trying to store the downloaded file and then trying to read that file
```
var downloadPath = path.resolve(__dirname, '../../data/file/downloaded');
    await client().send('Page.setDownloadBehavior', {
        behavior: 'allow',
        downloadPath: downloadPath,
    });
    await click(link('Download'))
    expect(path.join(downloadPath, 'my_resume')).to.exist;
    pdfParser.loadPDF(await path.join(downloadPath, 'my_resume.pdf'));
```
I am getting error as no such file or directory, stat 'data\file\downloaded\my_resume.pdf' . Can I wait for that file to be stored on that? After the test failed, I can see that file is stored on that specified path. Is there any better way to do a certain task after completing some task?

**Comment by NivedhaSenthil** [comment]
you can use Taiko's [waitFor](https://docs.taiko.dev/api/waitfor/) api to wait for any predicate...for your case the predicate has be to check if file exists

Eg: `await waitFor(() => fs.existsSync(<path>))`

**Reply by ramesh-khatri** [reply]
@NivedhaSenthil  Thank you so much :)
